res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,452 @@
+    NumarkScratch.effect = new components.ComponentContainer();
+    for (var i = 0; i < 2; i++) {
+        NumarkScratch.deck[i] = new NumarkScratch.Deck(i + 1);
+        NumarkScratch.effect[i] = new NumarkScratch.EffectUnit(i + 1);
+    }
+
+    NumarkScratch.gains = new NumarkScratch.Gains();
+
+    //Toggle the vinyl control section in skins to show
+    engine.setValue("[VinylControl]", "show_vinylcontrol", 1);
+
+    // Send Serato SysEx messages to request initial state and unlock pads
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x00, 0xF7], 6);
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x01, 0xF7], 6);
+
+    engine.makeConnection("[Channel1]", "VuMeter", NumarkScratch.vuCallback);
+    engine.makeConnection("[Channel2]", "VuMeter", NumarkScratch.vuCallback);
+
+    // Trigger is needed to initialize lights to 0x01
+    NumarkScratch.deck.forEachComponent(function(component) {
+        component.trigger();
+    });
+    NumarkScratch.effect.forEachComponent(function(component) {
+        component.trigger();
+    });
+
+    // set FX buttons init light & Shift button init light
+    midi.sendShortMsg(0x98, 0x00, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x01, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x02, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x03, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x04, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x05, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x9F, 0x32, NumarkScratch.LOW_LIGHT);
+};
+
+NumarkScratch.shutdown = function() {
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x04, 0xF7], 6);
+};
+
+NumarkScratch.shift = function() {
+    NumarkScratch.shifted = true;
+    NumarkScratch.deck.shift();
+    NumarkScratch.effect.shift();
+};
+
+NumarkScratch.unshift = function() {
+    NumarkScratch.shifted = false;
+    NumarkScratch.deck.unshift();
+    NumarkScratch.effect.unshift();
+};
+
+NumarkScratch.allEffectOff = function() {
+    NumarkScratch.effect[0].effects=[false, false, false];
+    NumarkScratch.effect[1].effects=[false, false, false];
+    NumarkScratch.FxUpdateLEDs();
+    NumarkScratch.effect[0].updateEffects();
+    NumarkScratch.effect[1].updateEffects();
+};
+
+NumarkScratch.FxUpdateLEDs = function() {
+    var newStates1=[false, false, false];
+    var newStates2=[false, false, false];
+    newStates1=NumarkScratch.effect[0].effects;
+    newStates2=NumarkScratch.effect[1].effects;
+    midi.sendShortMsg(0x98, 0x00, newStates1[0] ? 0x7F:NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x01, newStates1[1] ? 0x7F:NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x02, newStates1[2] ? 0x7F:NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x03, newStates2[0] ? 0x7F:NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x04, newStates2[1] ? 0x7F:NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x05, newStates2[2] ? 0x7F:NumarkScratch.LOW_LIGHT);
+};
```

Comment:
please clean this up and use proper OOP principles. Create a component instance for each control of the controller and communicate through those. So create a `components.Button` for each FX Select Button, A custom Button for the "FX-paddles", etc.